src/main/java/com/blackducksoftware/integration/jira/config/controller/action/IssueCreationConfigActions.java

```diff
@@ -208,9 +216,33 @@ public BlackDuckJiraConfigSerializable updateConfig(final BlackDuckJiraConfigSer
         globalConfigurationAccessor.setIssueCreationConfig(issueCreationConfig);
         updatePluginTaskInterval(previousInterval.orElse(0), intervalBetweenChecks);
 
+        startUserAssignment();
+
         return config;
     }
 
+    private void startUserAssignment() {
+        if (null != assignmentThread && assignmentThread.isAlive()) {
+            assignmentThread.interrupt();
+        }
+        assignmentThread = new Thread("userAssignmentThread") {
```

Comment:
The use of a Future here would be able to handle what you want to do with a timeout.  This seems like this code matches what you messaged me about privately.  So you should be able to create a Future object with an executor service and use the get(300, TimeUnit.SECONDS) method of the future.  It will throw a TimeoutException if the timeout has occurred.